lib/galaxy/datatypes/sequence.py

```diff
@@ -336,8 +336,43 @@ def split(cls, input_datasets: List, subdir_generator_function: Callable, split_
         raise NotImplementedError("Can't split generic alignment files")
 
 
+class BaseSequence(Sequence):
```

Comment:
The class names should be the other way around, Base* something should be the thing you inherit from. I don't think there's a reason you could not push the display_data method right into the current Sequence class (and replace `Sequence.display_data(self, trans, dataset, preview, filename, to_ext, **kwd)` with `super().display_data(self, trans, dataset, preview, filename, to_ext, **kwd)`)

Reply:
Thanks for comment, I agreed and modified accordingly. As I understand, there is no need to pass self argument when using super() because it automatically passes self. But I don't know why the tests still use the version where self was explicitly passed, which leads to argument mismatches and results errors "too many arguments" in the method call. Removing self should resolve these errors.

Reply:
Now all seems to be fine